packages/roosterjs-content-model/lib/modelApi/block/setModelAlignment.ts

```diff
@@ -0,0 +1,77 @@
+import { alignTable } from '../table/alignTable';
+import { ContentModelDocument } from '../../publicTypes/group/ContentModelDocument';
+import { ContentModelListItem } from '../../publicTypes/group/ContentModelListItem';
+import { getOperationalBlocks } from '../selection/collectSelections';
+import { isBlockGroupOfType } from '../common/isBlockGroupOfType';
+import { TableOperation } from 'roosterjs-editor-types';
+
+const ResultMap: Record<
+    'left' | 'center' | 'right',
+    Record<'ltr' | 'rtl', 'start' | 'center' | 'end'>
+> = {
+    left: {
+        ltr: 'start',
+        rtl: 'end',
+    },
+    center: {
+        ltr: 'center',
+        rtl: 'center',
+    },
+    right: {
+        ltr: 'end',
+        rtl: 'start',
+    },
+};
+
+const TableAlignMap: Record<
+    'left' | 'center' | 'right',
+    Record<
+        'ltr' | 'rtl',
+        TableOperation.AlignLeft | TableOperation.AlignCenter | TableOperation.AlignRight
+    >
+> = {
+    left: {
+        ltr: TableOperation.AlignLeft,
+        rtl: TableOperation.AlignRight,
+    },
+    center: {
+        ltr: TableOperation.AlignCenter,
+        rtl: TableOperation.AlignCenter,
+    },
+    right: {
+        ltr: TableOperation.AlignRight,
+        rtl: TableOperation.AlignLeft,
+    },
+};
+
+/**
+ * @internal
+ */
+export function setModelAlignment(
+    model: ContentModelDocument,
+    alignment: 'left' | 'center' | 'right'
+) {
+    const paragraphOrListItemOrTable = getOperationalBlocks<ContentModelListItem>(
+        model,
+        ['ListItem'],
+        []
+    );
+
+    paragraphOrListItemOrTable.forEach(({ block }) => {
+        if (isBlockGroupOfType<ContentModelListItem>(block, 'ListItem')) {
```

Comment:
save the result of `ResultMap[alignment][format.direction == 'rtl' ? 'rtl' : 'ltr']` to a local var since it is used multiple times.

I think the handling for table can also be merged together. But that can be done in separate PR.